Sources/Paywalls/Components/Common/PartialComponent.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PartialComponent.swift
+//
+//  Created by Josh Holtz on 10/26/24.
+//
+// swiftlint:disable missing_docs
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+public extension PaywallComponent {
+
+    protocol PartialComponent: PaywallComponentBase {
+
+        var visible: Bool? { get }
+
+    }
+
+    struct ComponentState<T: PartialComponent>: PaywallComponentBase {
+
+        public init(selected: T?, introOffer: T?) {
+            self.selected = selected
+            self.introOffer = introOffer
+        }
+
+        public let selected: T?
+        public let introOffer: T?
+
+    }
+
+    enum ComponentConditionsType {
+        case mobileLandscape, tablet, tabletLandscape, desktop
+    }
```

Comment:
Same comment on the devices as https://github.com/RevenueCat/purchases-ios/pull/4414#discussion_r1819385459.